src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1833,10 +1836,41 @@ internal ImmutableArray<BoundNode> GetBoundNodes(CSharpSyntaxNode node)
             var statementBinder = GetEnclosingBinder(GetAdjustedNodePosition(nodeToBind));
             Binder incrementalBinder = new IncrementalBinder(this, statementBinder);
 
+            // PROTOTYPE(nullable-api): Speculative models are going to have to do something more advanced
+            // here. They will need to run nullable analysis up to the point that is being speculated on, and
+            // then take that state and run analysis on the statement or expression being speculated on.
+            // Currently, it will return incorrect info because it's just running analysis on the speculated
+            // part.
+
+            BoundNode root;
             using (_nodeMapLock.DisposableWrite())
             {
-                BoundNode boundStatement = this.Bind(incrementalBinder, nodeToBind, _ignoredDiagnostics);
-                results = GuardedAddBoundTreeAndGetBoundNodeFromMap(node, boundStatement);
+                root = this.Bind(incrementalBinder, nodeToBind, _ignoredDiagnostics);
+            }
+
+            if (EnableNullableAnalysis)
+            {
+                // PROTOTYPE(nullable-api): Handle other symbol types
+                if (MemberSymbol is MethodSymbol method)
+                {
+                    root = NullableWalker.AnalyzeAndRewrite(Compilation, method, root, RootBinder.Conversions, _ignoredDiagnostics);
+                }
+            }
+#if DEBUG
+            else
+            {
+                // Always rewrite in DEBUG mode and just throw the results away.
+                if (MemberSymbol is MethodSymbol method)
+                {
+                    _ = NullableWalker.AnalyzeAndRewrite(Compilation, method, Bind(incrementalBinder, Root, _ignoredDiagnostics), RootBinder.Conversions, _ignoredDiagnostics);
```

Comment:
>_ignoredDiagnostics [](start = 108, length = 19)

Perhaps create a `new DiagnosticBag()` for this codepath, to minimize the observable differences in `DEBUG`. (Even though the current `DiagnosticBag` is named `_ignoredDiagnostics`, it is stored in a field so some member might look at the value later.)